lib/pos_random_seed.go

```diff
@@ -0,0 +1,182 @@
+package lib
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"github.com/deso-protocol/core/bls"
+	"github.com/dgraph-io/badger/v3"
+	"github.com/golang/glog"
+	"github.com/holiman/uint256"
+	"github.com/pkg/errors"
+	"io"
+)
+
+//
+// TYPES: RandomSeedHash
+//
+
+type RandomSeedHash [32]byte
+
+func (randomSeedHash *RandomSeedHash) ToUint256() *uint256.Int {
+	return uint256.NewInt().SetBytes(randomSeedHash.ToBytes())
+}
+
+func (randomSeedHash *RandomSeedHash) Copy() *RandomSeedHash {
+	randomSeedHashCopy := &RandomSeedHash{}
+	copy(randomSeedHashCopy[:], randomSeedHash[:])
+	return randomSeedHashCopy
+}
+
+func (randomSeedHash *RandomSeedHash) Eq(other *RandomSeedHash) bool {
+	return bytes.Equal(randomSeedHash.ToBytes(), other.ToBytes())
+}
+
+func (randomSeedHash *RandomSeedHash) ToBytes() []byte {
+	return randomSeedHash[:]
+}
+
+func (randomSeedHash *RandomSeedHash) FromBytes(randomSeedHashBytes []byte) (*RandomSeedHash, error) {
+	if len(randomSeedHashBytes) != 32 {
+		return nil, errors.Errorf("RandomSeedHash.FromBytes: input has length %d but should have length 32", len(randomSeedHashBytes))
+	}
+	copy(randomSeedHash[:], randomSeedHashBytes)
+	return randomSeedHash, nil
+}
+
+func EncodeRandomSeedHash(randomSeedHash *RandomSeedHash) []byte {
+	return EncodeByteArray(randomSeedHash.ToBytes())
+}
+
+func DecodeRandomSeedHash(rr io.Reader) (*RandomSeedHash, error) {
+	randomSeedHashBytes, err := DecodeByteArray(rr)
+	if err != nil {
+		return nil, errors.Wrapf(err, "DecodeRandomSeedHash: problem reading RandomSeedHash from bytes: ")
+	}
+	return (&RandomSeedHash{}).FromBytes(randomSeedHashBytes)
+}
+
+//
+// UTXO VIEW UTILS
+//
+
+func (bav *UtxoView) GenerateRandomSeedSignature(signerPrivateKey *bls.PrivateKey) (*bls.Signature, error) {
```

Comment:
Include a lot of comments explaining that this signature will be deterministic etc...